### PR TITLE
builder/azure-arm: Make tenant_id optional

### PR DESCRIPTION
**Pull request by paulmey**
Look up tenant id if not specified by the user. Reusing the code that was used by device code flow.

PTAL: @boumenot

### CODE REVIEW

File: website/source/docs/builders/azure.html.md [hunk 1]
```diff
@@ -72,6 +70,8 @@ builder.
 -   `image_url` (string) Specify a custom VHD to use.  If this value is set, do not set image_publisher, image_offer,
      image_sku, or image_version.
 
+-   `tenant_id` (string) The account identifier with which your `client_id` and `subscription_id` are associated.
```
Comment (boumenot) [inline website/source/docs/builders/azure.html.md]: Consider adding a blurb that the value is looked up if not set.

File: builder/azure/arm/config_retriever_test.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+	if err := sut.FillParameters(c); err != nil {
+		t.Errorf("Unexpected error when calling sut.FillParameters: %v", err)
+	}
+
+	if expected := tid; c.TenantID != expected {
+		t.Errorf("Expected TenantID to be %q but got %q", expected, c.TenantID)
+	}
+
+	errorString := "error while retrieving tenant-id"
```
Comment (boumenot) [inline builder/azure/arm/config_retriever_test.go]: NIT: tenant_id or tenant-id or TenantID - pick one.

File: builder/azure/arm/config_retriever_test.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package arm
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+func TestConfigRetrieverFillsTenantIDWhenEmpty(t *testing.T) {
```
Comment (boumenot) [inline builder/azure/arm/config_retriever_test.go]: I would break this into more explicit test cases.

**Comment by boumenot** [comment]
Few nits, otherwise LGTM.

**Comment by cbednarski** [comment]
This looks great! Thanks for updating the docs, too.

**Pull request merged**